src/apps/service/apps/job_listings.clj

```diff
@@ -68,18 +72,31 @@
     :batch           (:is_batch job)
     :batch_status    (when (:is_batch job) (format-batch-status id))}))
 
+(defn- interactive-urls
+  [job rep-steps]
+  (let [user-id (get-valid-user-id (job :username))
```

Comment:
Typically it's safer to access a map's value with `(:key map)`, in case the map is `nil`.
Using a `nil` map as the function will throw an exception, but using the `:key` as the function with a `nil` argument will not.